src/Host/Client/Impl/Session/RSession.cs

```diff
@@ -38,7 +37,7 @@ internal sealed class RSession : IRSession, IRCallbacks {
         public event EventHandler<EventArgs> Disconnected;
         public event EventHandler<EventArgs> Disposed;
         public event EventHandler<EventArgs> DirectoryChanged;
-        public event EventHandler<EventArgs> BeforePackagesInstalled;
+        public event AsyncEventHandler<EventArgs> BeforePackagesInstalledAsync;
```

Comment:
OK. What's wrong with them? Currently there is only one listener.  Async call is on `IRCallbacks.BeforePackagesInstalledAsync`. Do we want then to add a method to `IRSessionCallback/RSessionCallback` and call the index directly?